util/qartod_qc_executor.py

```diff
@@ -0,0 +1,185 @@
+import os
+import io
+import json
+import importlib
+import logging
+from util.qartod_service import qartodTestServiceAPI
+
+import numpy as np
+
+from util.common import QartodFlags
+
+log = logging.getLogger(__name__)
+
+EXCEPTION_MESSAGE = "Logged Exception"
+
+# QARTOD Parameter identification patterns
+QARTOD_PRIMARY = 'qartod_flag_primary'
+QARTOD_SECONDARY = 'qartod_flag_secondary'
+
+
+class QartodQcExecutor(object):
+    """
+    Class to manage the execution of all necessary QC for a request
+    """
+    def __init__(self, stream_request):
+        self.stream_request = stream_request
+        self.request_id = stream_request.request_id
+
+    def execute_qartod_tests(self):
+        for stream_key, stream_dataset in self.stream_request.datasets.iteritems():
+            subsite, node, sensor, _, stream = stream_key.as_tuple()
+            parameters = [parameter.name for parameter in stream_key.stream.parameters]
+            qartod_tests = qartodTestServiceAPI.find_qartod_tests_bulk(subsite, node, sensor, stream, parameters)
+
+            for dataset in stream_dataset.datasets.itervalues():
+                for qartod_test_record in qartod_tests:
+                    self.execute_qartod_test(qartod_test_record, dataset)
+
+    def execute_qartod_test(self, qartod_test_record, dataset):
+        parameter = qartod_test_record.parameter
+        test_identifier = qartod_test_record.test
+
+        # if the full function path is specified, parse the module from the path; otherwise, assume the function is from
+        # the default module and the test identifier is simply the function name
+        if test_identifier.count(".") > 0:
+            module, test = test_identifier.rsplit('.', 1)
+        else:
+            module = "ion_functions.qc.qartod_functions"
+            test = test_identifier
+
+        if parameter not in dataset:
```

Comment:
Looks like this "if" block should be moved before the "if test_identifier.count" code block.